### PR TITLE
Python: Start 1D Support

### PR DESCRIPTION
**Pull request by ax3l**
Start supporting 1D builds in Python.

Follow-up to #2310 
Needs #2586

### CODE REVIEW

**Comment by ax3l** [comment]
- [x] First `-Wunused` error:
```
  /home/runner/work/WarpX/WarpX/Source/BoundaryConditions/PML.cpp:114:17: error: unused function 'FillZero' [-Werror,-Wunused-function]
      static void FillZero (int idim, Sigma& sigma, Sigma& sigma_cumsum,
                  ^
  1 error generated.
```

**Comment by ax3l** [comment]
@RemiLehe @prkkumar ready for review/merge :)

File: Source/Filter/NCIGodfreyFilter.cpp [hunk 1]
```diff
@@ -23,11 +23,12 @@
 using namespace amrex;
 
 NCIGodfreyFilter::NCIGodfreyFilter(godfrey_coeff_set coeff_set, amrex::Real cdtodz, bool nodal_gather){
-#if (AMREX_SPACEDIM >= 2)
     // Store parameters into class data members
     m_coeff_set = coeff_set;
     m_cdtodz = cdtodz;
     m_nodal_gather = nodal_gather;
+
+#if (AMREX_SPACEDIM >= 2)
```
Comment (prkkumar) [inline Source/Filter/NCIGodfreyFilter.cpp]: I think, Lines 31-43 can be simplified with 
```c++
#if (AMREX_SPACEDIM == 3)
    stencil_length_each_dir = {1,1,5};
    slen = {1,1,5};
#elif (AMREX_SPACEDIM == 2)
    stencil_length_each_dir = {1,5};
    slen = {1,5,1};
#else
    amrex::ignore_unused(coeff_set, cdtodz, nodal_gather);
    amrex::Abort("NCIGodfreyFilter not implemented in 1D!");
#endif
```

Comment (ax3l) [reply Source/Filter/NCIGodfreyFilter.cpp:31]:
```suggestion
```

Comment (ax3l) [reply Source/Filter/NCIGodfreyFilter.cpp]: yep, haha what did I do here :D

**Review by prkkumar — APPROVED** [review]

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**